### PR TITLE
Quickstart ordering

### PR DESCRIPTION
**Pull request by okjake**

### CODE REVIEW

File: src/components/platform-list.jsx [hunk 1]
```diff
@@ -2,11 +2,35 @@ import React from 'react';
...
+  constructor() {
+    super();
+    this.comparator = this.comparator.bind(this);
+  }
+
+  comparator(a, b) {
+    // show third party quickstarts last
+    const thirdParty = [
+      this.props.quickstart.platforms[a].third_party,
+      this.props.quickstart.platforms[b].third_party,
```
Comment (nkohari) [inline src/components/platform-list.jsx]: Assigning the values into arrays makes this a little difficult to understand. Maybe something like this?

```js
const {platforms} = this.props.quickstart;
if (platforms[a].third_party !== platforms[b].third_party) { ... }
```
Then later on you can compare `platforms[a].title` and `platforms[b].title`.

File: src/components/platform-list.jsx [hunk 2]
```diff
@@ -2,11 +2,35 @@ import React from 'react';
...
+    ];
+
+    if (thirdParty[0] !== thirdParty[1]) {
+      return thirdParty[0] && !thirdParty[1] ? 1 : -1;
```
Comment (nkohari) [inline src/components/platform-list.jsx]: There's a lot of operator nesting here, which makes the logic hard to parse. Maybe just adding parentheses would help... they aren't technically necessary but would help the readability.

File: src/components/platform-list.jsx [hunk 3]
```diff
@@ -2,11 +2,35 @@ import React from 'react';
...
+    }
+
+    // sort alphabetically
+    const titles = [
+      this.props.quickstart.platforms[a].title.toLowerCase(),
+      this.props.quickstart.platforms[b].title.toLowerCase(),
+    ];
+
+    return titles[0] < titles[1] ? -1 : titles[0] > titles[1] ? 1 : 0;
```
Comment (nkohari) [inline src/components/platform-list.jsx]: Nesting the ternary operator makes this very difficult to understand. Also, it might be easiest just to use `localeCompare()`:

```js
platforms[a].title.localeCompare(platforms[b].title);
```

File: src/components/platform-list.jsx [hunk 4]
```diff
@@ -2,11 +2,35 @@ import React from 'react';
...
+  }
+
   render() {
     let {quickstart, customNavigationAction} = this.props;
-    
-    let items = Object.keys(quickstart.platforms).map((name, i) => (
+    let items = Object.keys(quickstart.platforms).sort(this.comparator).map((name, i) => (
```
Comment (nkohari) [inline src/components/platform-list.jsx]: This is probably just a personal preference, but it may make sense to nest `comparator()` as a function within `render()` rather than making it a member of the class itself. I tend to like to do that with predicate/filter functions like this, but again it's probably just my own bias. :)

**Comment by okjake** [comment]
Thanks Nate, all makes sense. I've added a new commit to hopefully make this more readable.

**Comment by nkohari** [comment]
Good stuff!
